libs/js-sdk/src/client.ts

```diff
@@ -44,13 +44,17 @@ export function createBrowserClient({
     if (store && initialValues) {
         throw new Error('Cannot specify both store and initialValues')
     }
-
-    const initialisedCallbacks: ((initialsed: boolean) => void)[] = []
-    const initialisedPromise = new PromiseCompletionSource<boolean>()
-    let updatedPromise: PromiseCompletionSource<boolean>[] = []
+    const initialPromise = new PromiseCompletionSource<boolean>()
+    const initialisedState: {
```

Comment:
JavaScript closures 'capture' variable references, this is problematic as this code uses closures quite a bit and expects to always have the latest variable reference available.

It's also one of the reasons the array was introduced.

Instead if you create a `state` object, then the closures capture a stable object reference but the values of the properties on that object can change.